### ISSUE TITLE
Update Bioformats

### ISSUE THREAD

**Issue by joaomamede** [soft-matter/pims#318]
We have a GE/Deltavision Ultra microscope in the lab and while the newest command line Bioformats works, pims still uses an old version.
Is there anyway of me updating the bioformats loci jar to the newest bioformats version?

Thanks,
JM

**Comment by GenevieveBuckley** [comment]
## Option 1 (quick to try)
In python, using the version of pims you have already:

```python
import pims
pims.bioformats._download_jar(version='6.0.1')  # the version number should be in quotes like this
pims.Bioformats('your_filename')
```

From the [list of bioformats versions](https://downloads.openmicroscopy.org/bio-formats/) it seems version 6.0.1 is the latest.


## Option 2 (also quick)
Instead of installing pims with the usual:
`pip install pims`

You could try:
`pip install git+https://github.com/caspervdw/pims.git@bump-bioformats`

This would give you loci_tools.jar version 5.8.2 (@caspervdw has an open pull request for this, so eventually this will make it into the master branch https://github.com/soft-matter/pims/pull/301)

## Option 3 (more complicated)
1. Make a fork of the pims repository (click the 'Fork' button in the top right hand corner of the pims page on github)
2. Make a new branch for your changes (you can do this by clicking the button 'branch: master' that is on the left hand side above the list of files in your repo. You could call the new branch `locitools`)
3. Edit line 57 of the file `pims/pims/bioformats.py`. Change the version number in quotes to the one you want:
```python
def _download_jar(version='5.7.0'):
```
4. Then reinstall pims with your new changes:
`pip install git+https://github.com/joaomamede/pims.git@locitools`

More generically, that's:
`pip install git+https://github.com/$YOUR_GITHUB_NAME/$GITHUB_REPO_NAME.git@$BRANCH_NAME`

**Comment by joaomamede** [comment]
Hi, Thank you for the help.

I tried the first option (that would work for me), however.
 `In [1]: import pims

In [2]: print pims.__version__
0.4.1

In [3]: pims.bioformats._download_jar(version='6.1.1')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-8b7609da9818> in <module>()
----> 1 pims.bioformats._download_jar(version='6.1.1')

AttributeError: 'module' object has no attribute '_download_jar'


In [4]: 
`

I guess I'll create a fork in github to try.

**Comment by joaomamede** [comment]
I just downloaded the latest loci_tools and put them in my ~/anaconda2/lib/python2.7/site-packages/pims

**Comment by GenevieveBuckley** [comment]
> I just downloaded the latest loci_tools and put them in my ~/anaconda2/lib/python2.7/site-packages/pims

Great, that should fix it :)

I didn't realize before, but it seems pims 0.4.1 was released just before [this commit](https://github.com/soft-matter/pims/commit/839a3a34ea1043885ed644d6ffd16d831f433759#diff-9ed83e29abe865730992f56a2dc561b5) (which is the one that defines `_download_jar`).

**Comment by nkeim** [comment]
As we prepare a new release of pims, what's the status of this (presumably never-ending) issue? Has anyone been using pims with newer versions of Bioformats like 6.3 or 6.4?

**Comment by nkeim** [comment]
FWIW, tests pass with Bioformats 6.4.0, so I'm leaning toward updating pull request #301 with that version and merging for the PIMS v0.5 release.

**Comment by GenevieveBuckley** [comment]
> FWIW, tests pass with Bioformats 6.4.0, so I'm leaning toward updating pull request #301 with that version and merging for the PIMS v0.5 release.

This sounds like a good plan. I'd also suggest adding something to the docs, so users know they *can* update bioformats/loci_tools.jar on their own. I'll open a PR with something similar to the comment I left above.

**Comment by rbnvrw** [comment]
This was solved in PR #301 and the Bioformats version we use now is 6.5.0. Thanks all for the good work :+1:
